coupons/models.py

```diff
@@ -100,14 +108,19 @@ def expired(self):
     @property
     def is_redeemed(self):
         """ Returns true is a coupon is redeemed (completely for all users) otherwise returns false. """
-        return self.users.filter(
-            redeemed_at__isnull=False
-        ).count() >= self.user_limit and self.user_limit is not 0
+        fully_redeemed_users = [
```

Comment:
This is a classic ORM impedance mismatch.

The property accesses a related model, so you are firing off a whole extra set of queries here.  You may decide it is worth it for the encapsulation, but good to make that decision explicitly if so.  Depending on the chain of relationships (no m2m...) a select_related might be an easy fix.

Reply:
Updated